### PR TITLE
REFACTOR: User system

### PR DESCRIPTION
**Pull request by theophile-jr**
* User is now Database as it contains more data
* Passwords are hashed in db
* Better error handling
* leaderboard
* handle user settings
image

### CODE REVIEW

**Review by Peralban — APPROVED** [review]

File: Engine/Server/Src/main.cpp [hunk 1]
```diff
@@ -6,6 +6,9 @@
 */
 
 #include "Application/Application.hpp"
+// #include "Database.hpp"
+// #include "Scores.hpp"
+// #include "Users.hpp"
```
Comment (mathieurobert1) [inline Engine/Server/Src/main.cpp:11]: You know.

File: Network/Database/Database.hpp [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+         */
+        ~Database() = default;
+
+        /**
+         * @brief Create the required tables for the database.
+         *
+         * @return true
+         * @return false
+         */
+        // bool createTables();
```
Comment (mathieurobert1) [inline Network/Database/Database.hpp:45]: Remove this if it's useless.

**Review by mathieurobert1 — CHANGES_REQUESTED** [review]
Some changes.
And also, why did you use so many raw pointers ? Is'nt it possible to pass them into shared_ptr ?

**Review by mathieurobert1 — APPROVED** [review]

File: Engine/Server/Src/main.cpp [hunk 3]
```diff
@@ -6,6 +6,10 @@
 */
 
 #include "Application/Application.hpp"
+#include "Database.hpp"
+#include "Scores.hpp"
+#include "Users.hpp"
```
Comment (mathieurobert1) [inline Engine/Server/Src/main.cpp:11]: Why did you need that on the main ?

**Review by mathieurobert1 — DISMISSED** [review]
Thanks for shared ptr

**Review by Peralban — APPROVED** [review]

**Review by Peralban — APPROVED** [review]

**Review by mathieurobert1 — APPROVED** [review]

**Pull request merged**